themes/default/content/blog/safe-lambda-secrets/index.md

```diff
@@ -0,0 +1,545 @@
+{{% /notes %}}
+
+### Deployment time
+
+Now that we've got everything together we can deploy this:
+
+```bash
+pulumi up
+```
+
+and you should see something like the following:
+
+```bash
+Previewing update (dev)
+
+View Live: https://app.pulumi.com/acmecorp/lambda-secrets/dev/previews/209f07fb-190d-4596-af9f-d2d72e9c5cc9
+
+     Type                                 Name                   Plan
+ +   pulumi:pulumi:Stack                  lambda-secrets-dev     create
+ +   ├─ aws:secretsmanager:Secret         secret1                create
+ +   ├─ aws:secretsmanager:Secret         secret2                create
+ +   ├─ aws:iam:Role                      roleLambdaWithSecrets  create
+ +   ├─ aws:secretsmanager:SecretVersion  secretversion-secret2  create
+ +   ├─ aws:secretsmanager:SecretVersion  secretversion-secret1  create
+ +   ├─ aws:iam:Policy                    secretsPolicy          create
+ +   ├─ aws:iam:RolePolicyAttachment      rpa-basic              create
+ +   ├─ aws:iam:RolePolicyAttachment      rpa-secrets            create
+ +   └─ aws:lambda:Function               lambdaWithSecrets      create
+
+Outputs:
+    lambdaName: "lambdaWithSecrets-b8ac5e9"
+
+Resources:
+    + 10 to create
+
+Do you want to perform this update? yes
+Updating (dev)
+
+View Live: https://app.pulumi.com/acmecorp/lambda-secrets/dev/updates/1
+
+     Type                                 Name                   Status
+ +   pulumi:pulumi:Stack                  lambda-secrets-dev     created
+ +   ├─ aws:secretsmanager:Secret         secret1                created
+ +   ├─ aws:secretsmanager:Secret         secret2                created
+ +   ├─ aws:iam:Role                      roleLambdaWithSecrets  created
+ +   ├─ aws:secretsmanager:SecretVersion  secretversion-secret2  created
+ +   ├─ aws:secretsmanager:SecretVersion  secretversion-secret1  created
+ +   ├─ aws:iam:Policy                    secretsPolicy          created
+ +   ├─ aws:iam:RolePolicyAttachment      rpa-basic              created
+ +   ├─ aws:lambda:Function               lambdaWithSecrets      created
+ +   └─ aws:iam:RolePolicyAttachment      rpa-secrets            created
+
+Outputs:
+    lambdaName: "lambdaWithSecrets-ca62cf8"
+
+Resources:
+    + 10 created
+
+Duration: 24s
+```
+
+Let's see what this has done. If we go to look at the Lambda Function in the console, click on the `Configuration` tab and select `Environment variables` you'll see the two plaintext environment variables and their values along with the two secret environment variables containing the ARNs to the relative secret in the Secrets Manager. Since we know what the environment variable key is (it's the uppercase version of what's in the stack config), it means we can reliably use that same name in the function code (the two lines where we call the `getSecretForLambda` function).
```

Comment:
```suggestion
Let's see what this has done. If we go to look at the Lambda Function in the console, click on the `Configuration` tab and select `Environment variables` you'll see the two plaintext environment variables and their values along with the two secret environment variables containing the ARNs to the relative secret in the Secrets Manager. Since we know the environment variable key (the uppercase version of what's in the stack config), we can reliably use that same name in the function code (the two lines where we call the `getSecretForLambda` function).
```